jclklib/client/notification_msg.cpp

```diff
@@ -62,8 +62,10 @@ void ClientNotificationMessage::addClientState(ClientState *newClientState)
 
 void ClientNotificationMessage::deleteClientState(ClientState *newClientState)
 {
-    ClientStateArray.erase(std::remove(ClientStateArray.begin(),
-        ClientStateArray.end(), newClientState), ClientStateArray.end());
+    auto it = std::remove(ClientStateArray.begin(), ClientStateArray.end(),
+                          newClientState);
```

Comment:
Just curious why the change is needed ?

Reply:
This change is to fix the error that found in CI, as shown in below. But this change does not solve the issue, I am working on another solution.

notification_msg.cpp: In static member function ‘static void JClkLibClient::ClientNotificationMessage::deleteClientState(JClkLibClient::ClientState*)’:
notification_msg.cpp:65:49: error: cannot convert ‘std::vector<JClkLibClient::ClientState*>::iterator’ to ‘const char*’
   65 |     auto it = std::remove(ClientStateArray.begin(), ClientStateArray.end(),
      |                           ~~~~~~~~~~~~~~~~~~~~~~^~
      |                                                 |
      |                                                 std::vector<JClkLibClient::ClientState*>::iterator
In file included from /usr/include/c++/14.1.1/cstdio:42,
                 from /usr/include/c++/14.1.1/ext/string_conversions.h:45,
                 from /usr/include/c++/14.1.1/bits/basic_string.h:4154,
                 from /usr/include/c++/14.1.1/string:54,
                 from /__w/libptpmgmt_iaclocklib/libptpmgmt_iaclocklib/jclklib/client/../client/jclk_client_state.hpp:19,
                 from notification_msg.cpp:15:
/usr/include/stdio.h:158:32: note:   initializing argument 1 of ‘int remove(const char*)’
  158 | extern int remove (const char *__filename) __THROW;
      |                    ~~~~~~~~~~~~^~~~~~~~~~